code/datums/components/strong_pull.dm

```diff
@@ -55,3 +59,35 @@ Basically, the items they pull cannot be pulled (except by the puller)
 /datum/component/strong_pull/proc/on_no_longer_pulled(datum/source, atom/movable/last_puller)
 	SIGNAL_HANDLER
 	lose_strong_grip()
+
+
+/**
+* Security variant of the component used for restraining people rather than crates
+*/
+/datum/component/strong_pull/security
```

Comment:
We don't want component subtypes unless absolutely necessary.

Reply:
I don't particularly understand what we need to be subtyping for, what does this make easier?

Reply:
> I don't particularly understand what we need to be subtyping for, what does this make easier?

It re-uses some of the code so I thought either making it a subtype or making both of them subtypes of a more generic strong_pull component would be better, if we shouldn't be making them subtypes I can change into strong_pull_security rather than a subtype, should I keep them in the same file in that case too?

Reply:
They shouldn't be in the same file

Reply:
but then this is copypaste

Reply:
it looks like it is overriding nearly all of it is why i ask

Reply:
the only thing left from what i can tell is registering a pull signal